tests/v2/validation/upgrade/kubernetes_test.go

```diff
@@ -106,4 +110,23 @@ func (u *UpgradeKubernetesTestSuite) testUpgradeSingleCluster(clusterName, versi
 		u.T().Logf("[%v]: Validating updated cluster's nodepools kubernetes versions", clusterName)
 		validateNodepoolVersions(u.T(), client, updatedCluster, version, !isCheckingCurrentCluster)
 	}
+
+	err = nodestat.IsNodeReady(client, clusterMeta.ID)
+	require.NoError(u.T(), err)
+
+	clusterToken, err := clusters.CheckServiceAccountTokenSecret(client, clusterName)
+	require.NoError(u.T(), err)
+	assert.NotEmpty(u.T(), clusterToken)
+
+	podResults, podErrors := pods.StatusPods(client, clusterMeta.ID)
+	assert.NotEmpty(u.T(), podResults)
+	assert.Empty(u.T(), podErrors)
+
+	if psact == string(RancherPrivileged) || psact == string(RancherRestricted) {
+		err = psadeploy.CheckPSACT(client, clusterName)
+		require.NoError(u.T(), err)
+
+		_, err = psadeploy.CreateNginxDeployment(client, clusterMeta.ID, psact)
+		require.NoError(u.T(), err)
+	}
```

Comment:
A question here, this is applicable for the current provisioning tests as well: Does making PSACT checks before the pods' check cover more? Cause PSACT creates another deployment if enabled

Reply:
I'm going to add a check actually. If deployment name of `nginx` already exists, I'll change the deployment name. Just so that we don't even have a possibility of a name conflict.

Reply:
Yes that's what I mean, in upgrade HA, provisioning tests and this test are chained, I think it might make it fail

Reply:
Yeah, I have a new commit I will push in a moment that should address 😄

Reply:
Resolved. Check if `nginx` already exists and if it does, create a new one with the prefix `upgrade-`.